src/main/core/manager.rs

```diff
@@ -494,10 +497,13 @@ impl<'a> Manager<'a> {
         // since the scheduler was dropped, all workers should have completed and the global object
         // and syscall counters should have been updated
 
+        let mut sim_stats = SimStats::new();
```

Comment:
It would be nicer to create a `SimStats` object at the start of the experiment that internally stores the object and syscall counters. Then when those counters need to be updated, we call through to the counters stored inside of the `SimStats` instance.

That way we can remove the current counters that exist there, and we can avoid cloning at the end. So, `worker::with_global_syscall_counter` and `worker::with_global_object_counters` would become `worker::with_global_sim_stats`.

Reply:
This has been changed. Now we have the two public structs `LocalSimStats`, `SharedSimStats`, and a private struct `SimStatsForOutput`.